core/src/Streamly/Data/Array/Generic/Mut.hs

```diff
@@ -0,0 +1,63 @@
+-- |
+-- Module      : Streamly.Data.Array.Generic.Mut
+-- Copyright   : (c) 2020 Composewell Technologies
+-- License     : BSD3-3-Clause
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+module Streamly.Data.Array.Generic.Mut
+(
+    -- * Type
+    -- $arrayNotes
+      Array (..)
+
+    -- * Constructing and Writing
+    -- ** Construction
+    -- *** Uninitialized Arrays
+    , newArray
```

Comment:
The names in this module should be consistent with names in Streamly.Data.Array/Streamly.Data.Array.Mut .